core/src/main/java/jenkins/management/PluginsLink.java

```diff
@@ -72,13 +72,43 @@ public Category getCategory() {
         return Category.CONFIGURATION;
     }
 
-    @Restricted(NoExternalUse.class)
-    public int getUpdateCount() {
+    @Override
+    public Badge getBadge() {
         final UpdateCenter updateCenter = Jenkins.get().getUpdateCenter();
         if (!updateCenter.isSiteDataReady()) {
             // Do not display message during this page load, but possibly later.
-            return 0;
+            return null;
+        }
+        List<Plugin> plugins = updateCenter.getUpdates();
+        int size = plugins.size();
+        if (size > 0) {
+            String tooltip = null;
+            Badge.Severity severity = Badge.Severity.warning;
+            int securityFixSize = (int) plugins.stream().filter(plugin -> plugin.fixesSecurityVulnerabilities()).count();
+            if (size > 1) {
+                switch (securityFixSize) {
+                    case 0:
+                        tooltip = Messages.PluginsLink_updatesAvailable(size);
+                        break;
+                    case 1:
+                        tooltip = Messages.PluginsLink_oneSecurityUpdateAvailable(size);
+                        severity = Badge.Severity.danger;
+                        break;
```

Comment:
Interesting approach, I kinda like it. Less red is better when there's no immediate need to act, especially with a badge that is practically always visible.

FWIW I look forward to reading in docs about the interaction between admin monitors and badges here :)